clients/python/agentic-sandbox-client/k8s_agent_sandbox/connector.py

```
@@ -155,6 +155,46 @@ def _is_port_open(self, port: int) -> bool:
         except (socket.timeout, ConnectionRefusedError):
             return False
 
+    def _preflight_check_router_service(self):
+        """Validates the router service exists in the configured namespace before port-forwarding.
+
+        Raises SandboxPortForwardError with namespace context and a remediation hint if the
+        service is definitively absent. Transient kubectl failures (e.g. network blip, RBAC
+        not covering 'get') are logged and silently ignored so they don't block a valid tunnel.
+        """
+        try:
+            result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)